File: hashtable/1.go
package hashtable

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

	for index, el := range nums {
		element, isIn := m[target-el]

		if isIn && (el != target-el || index != element) {
			return []int{index, element}
		} else {
			m[el] = index
		}
	}

	return []int{0, 0}
}

File: array/1.py

from typing import List

def twoSum(nums: List[int], target: int) -> List[int]:
    d = dict()

    for index, num in enumerate(nums):
        if target - num in d:
            return [index, d[target - num]]
        else:
            d[num] = index
    
    print(d)



print(twoSum([3,3], 6))